compiler/aarch64/codegen/OMRTreeEvaluator.cpp

```diff
@@ -498,10 +499,69 @@ OMR::ARM64::TreeEvaluator::exceptionRangeFenceEvaluator(TR::Node *node, TR::Code
 
 TR::Register *
 OMR::ARM64::TreeEvaluator::loadaddrEvaluator(TR::Node *node, TR::CodeGenerator *cg)
-	{
-	// TODO:ARM64: Enable TR::TreeEvaluator::loadaddrEvaluator in compiler/aarch64/codegen/TreeEvaluatorTable.hpp when Implemented.
-	return OMR::ARM64::TreeEvaluator::unImpOpEvaluator(node, cg);
-	}
+   {
+   TR::Register *resultReg;
+   TR::Symbol *sym = node->getSymbol();
+   TR::Compilation *comp = cg->comp();
+   TR::MemoryReference *mref = new (cg->trHeapMemory()) TR::MemoryReference(node, node->getSymbolReference(), 4, cg);
```

Comment:
Can you explain what the constant `4` means for the length?

Reply:
The length can be any value in this case.
The result of this evaluator is the address pointed by this `MemoryReference`, and the length does not have particular meaning.

Reply:
OK.  Choosing "4" is very specific, and to the untrained eye implies some kind of purpose.  I would suggest either passing "0", or having the length parameter optional and defaulting to something sensible (like 0).

Reply:
Changed the length to 0.

Reply:
Actually, what is the `_length` field for in the memory reference?  Nothing uses it at the moment.  Is it even needed?

Reply:
The p version of `OMRMemoryReference` takes a length in its constructors, but it seems it is never referenced.  The arm32 version uses it in the `addToOffset()`.
I would like to revisit it as a future task.